lib/aliases.rb

```diff
@@ -2,7 +2,12 @@
 
 module Homebrew
   module Aliases
-    BASE_DIR = File.expand_path "~/.brew-aliases"
+    # Unix-Like systems store config in $HOME/.config whose location can be
+    # overriden by the XDG_CONFIG_HOME environment variable. Unfortunately
+    # Homebrew strictly filters environment variables in BuildEnvironment.
+    BASE_DIR = Pathname.new("~/.config/brew-aliases").expand_path
+    # Default to HOME to avoid creating .config in case XDG_CONFIG_HOME is set.
+    BASE_DIR = Pathname.new("~/.brew-aliases").expand_path unless BASE_DIR.directory?
```

Comment:
Playing with `brew irb` I found that it does remove `XDG_CONFIG_HOME` from the env, but it keeps every `HOMEBREW_*` variable. We can support e.g. `HOMEBREW_ALIASES_PATH` or `HOMEBREW_ALIASES_DIR`. I think this would be better than just checking for the existence of `~/.config`, because it may break things: on my machine I have both `~/.config` and `~/.brew-aliases`.

Reply:
Every developer has `~/.config`. It's the default `XDG_CONFIG_HOME`. It's where the types of things go you'd version in a dotfiles repo. Precisely like this project. This line specifically checks for `~/.config/brew-aliases`. I didn't make it default because we can't check if the variable is set. In practice I doubt anyone does and it isn't worth extra configuration.

Reply:
Thanks for the explanation. I thought having a `~/.config` would mess up my installation but the code does check for `~/.config/brew-aliases`, so it’s fine.

Could you just extract the code in some block or function so we have only one `BASE_DIR =` declaration?

Reply:
Do you like this?

```ruby
BASE_DIR = begin
   dir.realpath
 rescue
   Pathname.new("~/.brew-aliases").expand_path
end
```